pkg/controllers/management/feature/feature_handler.go

```diff
@@ -0,0 +1,100 @@
+package feature
+
+import (
+	"context"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	"github.com/rancher/rancher/pkg/features"
+	managementv3 "github.com/rancher/rancher/pkg/generated/controllers/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/wrangler"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type handler struct {
+	featuresClient managementv3.FeatureClient
+	tokensLister   managementv3.TokenCache
+	tokenEnqueue   func(string, time.Duration)
+}
+
+func Register(ctx context.Context, wContext *wrangler.Context) {
+	h := handler{
+		featuresClient: wContext.Mgmt.Feature(),
+		tokensLister:   wContext.Mgmt.Token().Cache(),
+		tokenEnqueue:   wContext.Mgmt.Token().EnqueueAfter,
+	}
+	wContext.Mgmt.Feature().OnChange(ctx, "feature-handler", h.sync)
+}
+
+func (h *handler) sync(_ string, obj *v3.Feature) (*v3.Feature, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
```

Comment:
Locked value seems like a field that should be on status, especially since we have no intention of allowing a user to change it. I don't think putting it on spec would remove the need for the locked value logic unless we counted on the UI to change it which would mean the API wouldn't set it. If we set it in the store we would need logic just like this. This also works if the value is set by kubectl.